api/app/Models/PoolCandidateSearchRequest.php

```diff
@@ -87,6 +87,49 @@ public static function scopeSearchRequestStatus(Builder $query, ?array $searchRe
         return $query;
     }
 
+    public static function scopeStreams(Builder $query, ?array $streams): Builder
+    {
+        if (empty($streams)) {
+            return $query;
+        }
+
+        // streams is an array of PoolStream enums
+        $query->whereHas('applicantFilter', function ($query) use ($streams) {
+            $query->where(function ($query) use ($streams) {
+                foreach ($streams as $index => $stream) {
+                    $query->orWhereJsonContains('qualified_streams', $stream);
+                }
+            });
+        });
+        return $query;
+    }
+
+    public static function scopeDepartments(Builder $query, ?array $departmentIds): Builder
+    {
+        if (empty($departmentIds)) {
+            return $query;
+        }
+
+        $query->whereHas('department', function ($query) use ($departmentIds) {
+            Department::scopeDepartmentsByIds($query, $departmentIds);
+        });
+        return $query;
```

Comment:
Just wondering, why did you add a a scope for this one, but just wrote the `whereIn` query for classifications?

Reply:
I wanted to make both of them re-usable by various models, but the classifications one I couldn't get to work due to the chained `whereHas` throwing "ambiguous id" errors. So I undid it in https://github.com/GCTC-NTGC/gc-digital-talent/pull/6897/commits/27fc4aed0791b7b3e349c1de595dfec0fe0d9e7c

Reply:
Ah, got it

Reply:
Maybe someone else can figure it out some other day